### PR TITLE
Make TemporalMemory more readable

### PR DESCRIPTION
**Pull request by chetan51**
Fixes https://github.com/numenta/nupic/issues/2496.

### CODE REVIEW

**Comment by chetan51** [comment]
@scottpurdy

File: nupic/research/temporal_memory.py [hunk 1]
```diff
@@ -110,65 +110,26 @@ def __init__(self,
...
+    Updates member variables:
+      - `activeCells`     (set)
+      - `winnerCells`     (set)
+      - `activeSegments`  (set)
+      - `predictiveCells` (set)
+      - `matchingSegments`(set)
+      - `matchingCells`   (set)
+      - `connections`     (Connections)
+    """
+    prevPredictiveCells = self.predictiveCells
```
Comment (scottpurdy) [inline nupic/research/temporal_memory.py]: these seem unnecessary - Why not just use self. versions in this function?

Comment (chetan51) [inline nupic/research/temporal_memory.py]: To indicate that they now represent the previous values of these variables.

Comment (rhyolight) [inline nupic/research/temporal_memory.py]: :+1: much clearer to me

**Comment by scottpurdy** [comment]
:+1:

**Pull request merged**